lib/link.js

```diff
@@ -5,253 +5,30 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
-var url = require('url');
-
-var libmanta = require('libmanta');
-var restify = require('restify');
-
-var common = require('./common');
 require('./errors');
 
 
-
-///--- Helpers
-
-// From restify
-function sanitizePath(p) {
-
-    // Be nice like apache and strip out any //my//foo//bar///blah
-    p = p.replace(/\/\/+/g, '/');
-
-    // Kill a trailing '/'
-    if (p.lastIndexOf('/') === (p.length - 1) && p.length > 1)
-        p = p.substr(0, p.length - 1);
-
-    return (p);
-}
-
-
-
-///--- Routes
-//
-// We only have put handlers, as once a link is written in the system all the
-// object APIs "just work". (note that on creating a new link, we actually set
-// the type to 'object' -> you can distinguish that an object was created via
-// link by checking the createdFrom attribute in moray).
-//
-
-
-//-- PUT Handlers --//
-
-function parseLocation(req, res, next) {
-    if (!req.headers.location)
-        return (next(new LinkRequiredError()));
-
-    req.link = {
-        owner: {}
-    };
-
-    try {
-        req.link.path =
-            sanitizePath(url.parse(req.headers.location).pathname);
-
-        var re;
-
-        if (common.STOR_PATH.test(req.link.path)) {
-            re = common.STOR_PATH;
-        } else if (common.PUBLIC_STOR_PATH.test(req.link.path)) {
-            re = common.PUBLIC_STOR_PATH;
-        } else if (common.JOBS_STOR_PATH.test(req.link.path)) {
-            re = common.JOBS_STOR_PATH;
-        } else if (common.REPORTS_STOR_PATH.test(req.link.path)) {
-            re = common.REPORTS_STOR_PATH;
-        }
-
-        if (re) {
-            var params = re.exec(req.link.path);
-            req.link.owner.login = decodeURIComponent(params[1]);
-        }
-
-    } catch (e) {
-        return (next(new InvalidLinkError(req)));
-    }
-
-    req.log.debug({
-        link: req.link
-    }, 'parseLocation: done');
-    next();
-}
-
-
-function resolveOwner(req, res, next) {
+function snapLinksDisabled(req, res, next) {
```

Comment:
nit: If possible, use a function name that isn't *exactly* the same search string as the `opts.snapLinksDisabled` set on a request object.

Reply:
ok